nucleus/osgi-platforms/osgi-cli-interactive/src/main/java/org/glassfish/osgi/cli/interactive/LocalOSGiShellCommand.java

```diff
@@ -195,29 +197,18 @@ private String[] enhanceForTarget(String[] args) {
 
 
     private Terminal createTerminal() throws IOException, CommandException {
-        InputStream inputStream;
-        OutputStream outputStream;
-        if (file != null) {
+        if (!isInteractive()) {
             if (!file.canRead()) {
                 throw new CommandException("File: " + file + " can not be read");
             }
-            inputStream = new FileInputStream(file);
-            outputStream = new EmptyOutputStream();
-        } else {
-            System.out.println(STRINGS.get("multimodeIntro"));
-            inputStream = new FileInputStream(FileDescriptor.in);
-            outputStream = System.out;
-        }
 
-        TerminalBuilder builder = TerminalBuilder.builder().streams(inputStream, outputStream);
-
-        if (System.getenv("TERM") == null) {
-            Terminal terminal = builder.type("dumb").build();
-            terminal.echo(false);
-            return terminal;
+            return new ExternalTerminal(REMOTE_COMMAND, "dumb", new FileInputStream(file), new EmptyOutputStream(),
+                encoding != null ? Charset.forName(encoding) : Charset.defaultCharset());
```

Comment:
I would just extract this ternary to a variable and inverted it (negations are painful for tired brains :D )

Reply:
Done ))